crates/plume/src/utils.nr

```diff
@@ -0,0 +1,13 @@
+pub fn to_field<let N: u32>(bytes: [u8; N]) -> Field {
+    let mut as_field = 0;
+    let mut offset = 1;
+    for i in 0..N {
+        let mut index = i;
+        index = N - i - 1;
+        as_field += (bytes[index] as Field) * offset;
+        offset *= 256;
+    }
+    std::as_witness(as_field);
```

Comment:
Just curious, why need this line?

Reply:
This little guy shaves off 200k constraints out of the whole thing 😄 a team member found a bug in the compiler that was causing a huge blowout. Using `as_witness` forces the compiler to assign intermediate values, which makes it behave as it should

Reply:
wow, nice trick